### ISSUE TITLE
unpacking-non-sequence false positive when RHS is a function from pyi

### ISSUE THREAD

**Issue by wojdyr** [pylint-dev/pylint#9354]
### Bug description

```python
I get E0633: Attempting to unpack a non-sequence (unpacking-non-sequence)
when using stub files for C extension. Reproduced here:

unpack.py:

import module1

def unpack():
    _, b = module1.func()
    return b


module1/__init__.pyi:
```py
def func() -> tuple[int, int]:
    ...
```
```


### Configuration

_No response_

### Command used

```shell
pylint unpack.py
```

### Pylint output

```python
************* Module unpack
unpack.py:4:4: E0633: Attempting to unpack a non-sequence (unpacking-non-sequence)
```

### Expected behavior

no error

### Pylint version

```shell
pylint 3.1.0-dev0
astroid 3.0.2
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

### OS / Environment

Linux

**Comment by mschmitzer** [comment]
I'm seeing the same behavior with a stub file for a Cython extension: Apparently, Pylint takes the `...` body of the stub literally and thus complains about the return type of the function not matching the use (in my case it is `non-iterator-returned`). Changing the stub implementation to return the "correct" type makes the problem go away.

So I think the solution for this would be for Pylint to treat stub definitions specially and either assume that they return whatever is needed at the call site or to look at the return type annotation instead of the function body.
